module/support/logging/DataPlayback/src/DataPlayback.cpp

```diff
@@ -52,12 +58,15 @@ namespace support {
                     catch (const std::exception& ex) {
 
                         // If we reached the end of the file we either terminate or loop depending on the setting
-                        if (input_file.eof()) {
+                        if (!input_file || input_file->eof()) {
```

Comment:
```suggestion
                        if (!input_file || !input_file->good()) {
```

Reply:
In this case, I specifically want to check fo see if we have reached the end of the file

Reply:
Dont we also care if a non-recoverable error has occurred on the file we are currently reading?

Reply:
I guess the real question is, why is branch special? If we take the other branch, next loop around we will be back at this branch anyway?